### ISSUE TITLE
Docs missing type="module"

### ISSUE THREAD

**Issue by sidewayss** [color-js/elements#152]
I wanted to kick the tires on the `<color-picker>`, so I copied sample text from [this docs page sub-section](https://elements.colorjs.io/#cdn) and I created the simplest of HTML files:
```html
<!DOCTYPE html>
<html>
  <head>
    <script src="https://elements.colorjs.io/src/color-picker/color-picker.js"></script>
  </head>
  <body>
    <color-picker></color-picker>
  </body>
</html>
```
In Chrome and Firefox it throws similar errors. In Chrome it's "Uncaught SyntaxError: Cannot use import statement outside a module (at color-picker.js:1:1)". It's failing on the first line of the module.

I tried the same thing with `src="https://elements.colorjs.io/index.js"` and it's the same error, different module.

I just realized that the docs are missing the `type="module"` attribute setting in the `<script>`, and that those "docs" seem to be based on the readme file.

**Comment by sidewayss** [comment]
Your welcome. Yes I see it fixed in both places.

Are you open for feature requests? I have an app I'm about to release that effectively mimics a combination of `<color-swatch>` and `<color-picker>`.  Here is a stable, but slightly out-of-date version: https://sidewayss.com/raf/apps/color/ 

If you care, limited docs are [here](https://sidewayss.github.io/rAF/apps/#color), and the current source is [here](https://github.com/sidewayss/rAF/tree/main/apps/color). It's part of an advanced animation library that integrates Color.js without depending on it.  There are two other apps that create the preset timing functions in the top row's `Easy:`  and `Multi:` drop-downs (click on `Easy:` to drop-down the label and select `Multi:`).

Note the second and third rows in the form, labeled `Start:` and `End:`:
- For the swatch, I don't see a way to put the `<label>` and `<input>` before the swatch graphic, in this case to its left.
- I need `<color-picker>` to give me the raw serialized color string, not the interpolated `display()`. The idea of these two fields is to allow any Color.js color space, not just those supported by the current browser.
- `<input type="color">` is a button that pops up the picker itself and returns a string.  I know I can wrap it up myself, but it would be great if `<color-picker>` could operate that way, or if there was a subclass or sub-type that did.

Maybe you can point me to a way to do these? Or let me know if it's appropriate to submit enhancement requests.  Thanks!

**Comment by DmitrySharabin** [comment]
> Or let me know if it's appropriate to submit enhancement requests.

Please do it regardless! We would love to know how people use the color elements, what they build with them, and what functionality they lack.

> For the swatch, I don't see a way to put the `<label>` and `<input>` before the swatch graphic, in this case to its left.

If I get what you want to achieve right, then `flex-direction: row-reverse` applied to the `<color-swatch>` itself should help. See this codepen as an example: https://codepen.io/dmitrysharabin/pen/zYgbdwb

> I need `<color-picker>` to give me the raw serialized color string, not the interpolated `display()`. The idea of these two fields is to allow any Color.js color space, not just those supported by the current browser.

I believe we can also do that (well if I got your idea correctly). See https://codepen.io/dmitrysharabin/pen/vYoPJQg


> `<input type="color">` is a button that pops up the picker itself and returns a string.  I know I can wrap it up myself, but it would be great if `<color-picker>` could operate that way, or if there was a subclass or sub-type that did.

That might be a reasonable component request.

**Comment by sidewayss** [comment]
@DmitrySharabin - Thanks for the codepens.  They look good.  So the `colorPicker.color` and `swatch.value` are not limited in their color spaces? The docs use the word "interpolated", and I assumed it was limited to the displayable color spaces.

> That might be a reasonable component request.

In this case I either need to wrap it up in a `<dialog>` with a `<button>` that opens it, or leave my code alone.  I would be willing, even interested, in collaborating on a new component that wrapped the swatch and the picker into something that emulates `<input type="color">`.  It would also require a drop-down, or other list, for selecting the color space. I would prefer contributing to that to building a custom wrapper for myself. Emulating the standard browser color picker seems like an obvious idea for a component.

Later today I'll submit an issue (you don't have any labels, like "enhancement request", enabled) in which I provide a bullet-point sketch of requirements/design.

*fyi* - regarding your codepens.  Don't use `querySelector()` if you can avoid it.  It's very slow.  In this case [`getElementsByTagName()`](https://developer.mozilla.org/en-US/docs/Web/API/Element/getElementsByTagName) is definitely preferable, even though it returns a pseudo-array so you have to do this:
```js
const picker = document.getElementsByTagName("color-picker")[0];
```
`getElementsByClassName()` and of course `getElementById()` are also useful in this regard.

**Comment by sidewayss** [comment]
I realized that, based on your first codepen, I could slot my label and text input inside a `<color-swatch>`. Doing so would simplify some code.

In the process, I was surprised to see that `row-reverse` applies to the swatch graphic relative to the slotted elements, but not to the slotted elements themselves, as if the slots were in a separate flex container to which the property doesn't apply.  I assume that's because the swatch is a shadow DOM element vs the slotted DOM elements. But how does that work?  Isn't the `<color-swatch>` itself a flex container?  I'm setting `flex-direction` for that container, but the label lays out in the normal, non-reverse direction.

See this fork of your codepen, and note that `row-reverse` not only puts the label after the input, it causes the entire `<div>` to align as if `justify-content: flex-end`:
https://codepen.io/sidewayss/pen/ExqJaxe

**Comment by sidewayss** [comment]
@DmitrySharabin - There are no discussions available for this repo so this is the only way I know to reach you...

I have a working [picker in a dialog](https://github.com/sidewayss/rAF/tree/main/apps/color), and I added [a comment](https://github.com/color-js/elements/issues/158#issuecomment-2481685038) to the feature request I submitted pointing to it and its source. Getting the raw, serialized string turned out to be simpler than your pens.  See [here](https://github.com/sidewayss/rAF/blob/705b6ff2382c436f02dddecb036b944b52d91e38/apps/color/events.js#L176). `picker.color` falls back to `color.toString()`, which is exactly what I need.

What is the "correct" way to hide the text input in the picker? It's in a shadow DOM inside the shadow DOM, a swatch within the picker.  Note that when it's set to `display:none`, `#swatch`'s height is a couple of pixels short.  Once I know how to hide the input I'll be able to submit a proper issue on that. Thanks.

**Comment by DmitrySharabin** [comment]
> There are no discussions available for this repo

Could you please check now?

> I have a working [picker in a dialog](https://github.com/sidewayss/rAF/tree/main/apps/color), and I added [a comment](https://github.com/color-js/elements/issues/158#issuecomment-2481685038) to the feature request I submitted pointing to it and its source. Getting the raw, serialized string turned out to be simpler than your pens. See [here](https://github.com/sidewayss/rAF/blob/705b6ff2382c436f02dddecb036b944b52d91e38/apps/color/events.js#L176). `picker.color` falls back to `color.toString()`, which is exactly what I need.

Awesome! I'm glad you could figure it out.

> What is the "correct" way to hide the text input in the picker?

This should help:

```css
color-picker::part(details) {
	display: none;
}
```

Similarly, you can style (including totally hide) any part of the color-picker. In conjunction with `CSS_prop: revert` (e.g., `border-radius: revert`) or `all: unset`, you achieve any desired result.

The available parts are: `sliders`, `swatch-base`, `gamut`, `details`, `info`.

> See this fork of your codepen, and note that `row-reverse` not only puts the label after the input, it causes the entire `<div>` to align as if `justify-content: flex-end`: https://codepen.io/sidewayss/pen/ExqJaxe

This is expected and actually has nothing to do with `justify-content: flex-end`. `<div>` is a block element (if not changed explicitly) and takes (horizontally) all the space its parent provides. In your case, the whole body. Applying `flex-direction: row-reverse` to the `<div>` element, you align its children from right to left, and they “jump” to the right edge of the `<div>`. Add, for example, `width: fit-content` to the `<div>`, and you'll get the desired result.

**Comment by sidewayss** [comment]
> Could you please check now?

Yes, now I see Discussions.  Thanks.

> Awesome! I'm glad you could figure it out.

It doesn't include `<color-swatch>`. I used a `<div>` instead.  I still haven't figured out how to format the swatch to match my current page.  That's why I submitted #159.

> This should help:

Thanks, I'll give that a try.  So the input is available as a part of the outer shadow DOM even though it's inside a nested shadow DOM? I'll have to look more closely at how this works...

> >  See this fork of your codepen, and note that row-reverse not only puts the label after the input, it causes the entire <div> to align as if justify-content: flex-end: https://codepen.io/sidewayss/pen/ExqJaxe

> This is expected and actually has nothing to do with justify-content: flex-end.

I'm not surprised by the `<div>` behavior. I was referring to the ways `<color-swatch>` *doesn't* act like the `<div>`, the ways in which it is exceptional.  I'm not trying to format a `<div>`, I'm trying to format a `<color-swatch>`.  This was just one additional difference.
